src/flamegpu/gpu/CUDAEnsemble.cu

```diff
@@ -224,6 +226,29 @@ unsigned int CUDAEnsemble::simulate(const RunPlanVector &plans) {
         printf("Ensemble time elapsed: %fs\n", ensemble_elapsed_time);
     }
 
+    // Send Telemetry
+    if (config.telemetry) {
+        // Generate some payload items
+        std::map<std::string, std::string> payload_items;
+        payload_items["GPUDevices"] = flamegpu::util::detail::compute_capability::getDeviceNames(config.devices);
+        payload_items["SimTime(s)"] = std::to_string(ensemble_elapsed_time);
+        // generate telemetry data
+        std::string telemetry_data = flamegpu::io::Telemetry::generateTelemetryData("ensemble-run", payload_items);
+        // send
+        if (!flamegpu::io::Telemetry::sendTelemetryData(telemetry_data)) {
+            if ((config.verbosity > Verbosity::Quiet))
+                fprintf(stderr, "Warning: Usage statistics for CUDAEnsemble failed to send.\n");
+        }
+        // print
+        if ((config.verbosity >= Verbosity::Verbose)) {
+            fprintf(stdout, "Telemetry packet sent to '%s' json was: %s\n", flamegpu::io::Telemetry::TELEMETRY_ENDPOINT, telemetry_data.c_str());
+        }
+    } else {
+        // Occasional hinting of telemetry if not in use (and not Quiet and not testing mode)
```

Comment:
This should be always, once per binary (via namespace scoped static) or never. Not based on prng.

Reply:
Doing this once per binary execution with an anon namespace'd static used to track this.